compliance/azure/azure_untagged_resources/untagged_resources.pt

```diff
@@ -0,0 +1,323 @@
+      all_input_tags=all_input_tags+','+param_tag_key_lower[j];
+    }
+    // Iterate through all resources
+    _.each(ds_azure_resources, function(ar){
+      var id = ar.id ;
+      var rg_split = id.split('/') ;
+      var resource_group_name = rg_split[4] ;
+      var namespace = rg_split[6] ;
+      var resource_type = rg_split[7] ;
+      var resource_tags = JSON.stringify(ar.tags);
+      var isResMissedTag=false;
+      var all_include_tag='';
+      var all_missed_tag=''; 
+      var all_tags={};
+      if(!(_.isEmpty(resource_tags))){
+      all_tags=ar.tags;
+      }
+      for(var j=0; j < param_tag_key_lower.length; j++){		  
+       if(_.isEmpty(resource_tags) || (resource_tags.toLowerCase().indexOf(param_tag_key_lower[j].toLowerCase()))==-1){
+         isResMissedTag=true;
+         all_missed_tag=all_missed_tag+" "+param_tag_key_lower[j];
+        }else{
+         all_include_tag=all_include_tag+" "+param_tag_key_lower[j];
+       }
+      }
+
+    if(isResMissedTag){	  
+      result.push({
+        subscriptionId: ar["subscriptionId"],
+        subscriptionName : ar["subscriptionName"],
+        id: ar["id"],
+        region: ar["region"],
+        name: ar["name"],
+        resource_group: resource_group_name,
+        namespace: namespace,
+        resource_type: resource_type,
+        all_missed_tag: all_missed_tag,
+        all_include_tag: resource_tags,
+        all_tags: all_tags		
+        })
+      }		  
+    })	
+    result = _.sortBy(result, 'subscriptionName');
+    result = _.sortBy(result, 'resource_type');	
+    result = _.sortBy(result, 'name');
+
+EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_azure_resource" do
+  validate $ds_filtered_resources do
+    summary_template "{{ len data }} Azure Resources out of compliance."
+    escalate $esc_email
+    escalate $esc_tag_resources
+    check eq(size(data),0) 
+    export do
+      resource_level true
+      field "subscriptionName" do
+        label "Subscription Name"
+      end
+      field "resource_type" do
+        label "Resource Type"
+      end	  
+      field "name" do
+        label "Resource Name"
+      end	  
+      field "resource_group" do
+        label "Resource Group"
+      end
+      field "region" do
+        label "Azure Region"
+      end
+     field "id" do
+        label "Resource ID"
+      end
+      field "all_include_tag" do
+        label "Existing Tags"
+      end
+      field "all_missed_tag" do
+        label "Missing Tags"
+      end
+    end	
+  end
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "esc_email" do
+   email $param_email
+end
+
+escalation "esc_tag_resources" do
```

Comment:
escalation needs **automatic false** so it does run when the incident is created.